pkg/iter/sc_finder.go

```diff
@@ -125,7 +120,7 @@ func (m *SubclusterFinder) FindSubclusters(ctx context.Context, flags FindFlags,
 	// a sandboxed subcluster can only be one of those in the vdb so we skip this part if
 	// we are looking for sandboxed subclusters
 	if isMainCluster && (flags&FindNotInVdb != 0 || flags&FindExisting != 0) {
-		missingSts, err := m.FindStatefulSets(ctx, flags & ^FindInVdb)
+		missingSts, err := m.FindStatefulSets(ctx, flags & ^FindInVdb, vapi.MainCluster)
```

Comment:
Does this still make sense to pass in main cluster here? Any harm in removing the `isMainCluster`, getting in here for a sandbox and passing the sandbox parm here. It would keep this logic simple. No need for the comment at line 120-121.

Reply:
This was to avoid calling the client to fetch sts when we don't need, but I see what you mean, now that we have moved the filter in buildObjList let's just pass the sandbox name.